### ISSUE TITLE
Disallow unknown options

### ISSUE THREAD

**Issue by sciurius** [miyagawa/cpanminus#604]
`cpanm` should abort with a help message when it encounters unrecognized command line options.

**Comment by miyagawa** [comment]
I wish it did but that's literally a breaking change for the client usage which depends on the current behavior because Getopt::Long enables auto abbrev by default.

Apparently you can set POSIXLY_CORRECT=1 environment variable to disable this behavior.

**Comment by sciurius** [comment]
> cpanm does die if you give an unrecognized command line options

     % cpanm -l xx --no-test FFI::CheckLib
    Unknown option: no-test
    --> Working on FFI::CheckLib
    Fetching http://www.cpan.org/authors/id/P/PL/PLICEASE/FFI-CheckLib-0.26.tar.gz ... OK
    Configuring FFI-CheckLib-0.26 ... OK
    ... etc ...

`cpanm` loudly complains about not recognizing `no-test`, but continues nevertheless.

FYI: You can specify option `test` to Getopt::Long as `test!` and Getopt::Long will handle `--test`, `--notest`, `--no-test` for you.

**Comment by miyagawa** [comment]
> FYI: You can specify option test to Getopt::Long as test! and Getopt::Long will handle --test, --notest, --no-test for you.

Yep i am aware,  but we already have `--no-notest` because `--notest` and `-n` were added first.

> cpanm loudly complains about not recognizing no-test, but continues nevertheless.

You're right and that, I suppose, is fixable, although i'm sure it will also break some clients who are currently giving unknown options that are swallowed in the warnings right now.

**Comment by sciurius** [comment]
:)
